include/ac_npc_shop_master.h

```diff
@@ -3,16 +3,265 @@
 
 #include "types.h"
 #include "m_actor.h"
+#include "ac_npc.h"
+#include "m_mail_password_check.h"
+#include "m_private.h"
+
+#define aNSC_MAX_TICKETS 255
+
+#define aNSC_9C8_LEN 8
+#define aNSC_9D0_LEN 8
+
+#define SELL_BUY_RATIO 4
+#define aNSC_LOAN_MEDIUM 148000
+#define aNSC_LOAN_LARGE 398000
+#define aNSC_LOAN_UPPER 798000
+#define aNSC_LOAN_STATUE 0
+#define aNSC_LOAN_BASEMENT 49800
+
+
+typedef struct npc_shop_master_s NPC_SHOP_MASTER_ACTOR;
+
+typedef void (*aNSC_PROC)(NPC_SHOP_MASTER_ACTOR*, GAME_PLAY*);
+
+struct npc_shop_master_s {
+    /* 0x000 */ NPC_ACTOR npc_class;
+    /* 0x994 */ int action;
+    /* 0x998 */ int next_action;
+    /* 0x99C */ aNSC_PROC proc;
+    /* 0x9A0 */ u16 _9A0;
+    /* 0x9A2 */ u8 _9A2;
+    /* 0x9A3 */ u8 _9A3;
+    /* 0x9A4 */ u8 _9A4;
+    /* 0x9A6 */ s16 player_angle;
+    /* 0x9A8 */ u32 _9A8;
+    /* 0x9AC */ u32 value;
+    /* 0x9B0 */ u32 counter;
+    /* 0x9B4 */ mActor_name_t sell_item;
+    /* 0x9B6 */ u8 ut_x;
+    /* 0x9B7 */ u8 ut_y;
+    /* 0x9B8 */ u8 _9B8;
+    /* 0x9B9 */ u8 _9B9;
+    /* 0x9BA */ mActor_name_t order_item;
+    /* 0x9BC */ u16 _9BC;
+    /* 0x9BE */ u16 _9BE;
+    /* 0x9C0 */ int _9C0;
+    /* 0x9C4 */ int _9C4;
+    /* 0x9C8 */ u8 _9C8[aNSC_9C8_LEN];
+    /* 0x9D0 */ u8 _9D0[aNSC_9D0_LEN];
+    /* 0x9D8 */ u8 password_str[mMpswd_PASSWORD_STR_LEN];
+    /* 0x9F4 */ u16 _9F4;
+    /* 0x9F6 */ u16 password_flags;
+    /* 0x9F8 */ mMpswd_password_c password;
+};
+
+enum aNSC_action {
+    aNSC_ACTION_GIVE_BALLOON = 0x7,
+    aNSC_ACTION_MONEY_OVERFLOW = 0xd,
+    aNSC_ACTION_BUY_REFUSE = 0xf,
+    aNSC_ACTION_BUY_TRANSACT = 0x11,
+    aNSC_ACTION_TAKE_OFF_HANDS = 0x14,
+    aNSC_ACTION_BREAK_BAG = 0x16,
+    aNSC_ACTION_ANYTHING_ELSE = 0x17,
+    aNSC_ACTION_EXIT = 0x43,
+    aNSC_ACTION_NUM
+};
+
+enum aNSC_msg_no {
+    aNSC_MSG_WAIT = -0x1,
```

Comment:
Enums should only have constant values assigned to them in cases where it's necessary. Here, the only one necessary would be aNSC_MSG_WAIT since it's -1.